README.md

```diff
@@ -1,58 +1,146 @@
+
+```console
+$ external-dns --zone example-org --google-project example-project --dry-run=false
+```
+
+Check that ExternalDNS created the desired DNS record for your service and that it points to its load balancer's IP. Then try to resolve it.
+
+```console
+$ dig +short nginx.example.org.
+104.155.60.49
+```
+
+Now you can experiment and watch how ExternalDNS makes sure that your DNS records are configured as desired. Here are a couple of things you can try out:
+* Change the desired hostname by modifying the Service's annotation.
+* Recreate the Service and see that the DNS record will be updated to point to the new load balancer IP.
+* Add another Service to create more DNS records.
+* Remove Services to clean up your managed zone.
+
+The [tutorials](docs/tutorials/gke.md) section contains a more detailed example of how to setup ExternalDNS in your Google Container Engine cluster.
+
+# Roadmap
+
+ExternalDNS was built with extensibility in mind. Adding and experimenting with new DNS providers and sources of desired DNS records should be as easy as possible. In addition, it should also be possible to modify how ExternalDNS behaves, e.g. whether it should add but must never delete records.
+
+Furthermore, we're working on an ownership system that allows ExternalDNS to keep track of the records it created and will allow it to never modify records it doesn't have control over.
+
+Here's a rough outline on what is to come:
+
+### v0.1
+
+* Support for Google CloudDNS
+* Support for Kubernetes Services
+
+### v0.2
+
+* Support for AWS Route53
+* Support for Kubernetes Ingresses
+
+### v0.3
+
+* Support for AWS Route53 via ALIAS
+* Support for multiple zones
+* Ownership System
+
+### v1.0
+
+* Ability to replace Kops' [DNS Controller](https://github.com/kubernetes/kops/tree/master/dns-controller)
+* Ability to replace Zalando's [Mate](https://github.com/zalando-incubator/mate)
+* Ability to replace Molecule Software's [route53-kubernetes](https://github.com/wearemolecule/route53-kubernetes)
+
+The [FAQ](docs/faq.md) contains additional information and goes into more detail. Also have a look at [the milestones](https://github.com/kubernetes-incubator/external-dns/milestones) to the the current state.
+
+### Unclear topics
+
+* Support for CoreDNS
+* Support for record weights
+* Support for different behavioral policies
+* Support for Services with `type=NodePort`
+* Support for TPRs
+* Support for more advanced DNS record configurations
+
+# Contributing
+
+You can build ExternalDNS for your platform with `make build`. The binary will land at `build/external-dns`.
+
+ExternalDNS's sources of DNS records live in package [source](source). They implement the `Source` interface that has a single method `Endpoints` which returns the represented source's objects converted to `Endpoints`. Endpoints are just a tuple of DNS name and target where target can be an IP or another hostname.
```

Comment:
I don't think this description belongs here. I would keep `Contributing` minimal and only describing how to contribute to the project. I would then have a document describing "how to get into the code" if this is where you are aiming at.